### DISCUSSION TITLE
Eliminating duplicate object detections from a video source

### DISCUSSION THREAD

**Discussion by Ziad-Assem** [ultralytics/ultralytics#2344] [Q&A] [answered]
Hello! I currently have a custom model that can detect license plates. It works really well on images but when I use a video source, it keeps detecting the same image over 10 times. I tried adding the argument iou and conf to try and reduce the duplicates but to no avail. Any suggestions? Thanks!

**Comment by dh031200** [comment, marked as answer]
@Ziad-Assem 
How about saving an image when an object is tracked over a few frames?
I wrote the following code using yolov8's object tracking.
Duplicate images of the same object can be suppressed.

[dh031200/ObjectExtrator](https://github.com/dh031200/ObjectExtractor)

**Reply by Ziad-Assem** [reply]
Thank you! I have a couple of questions:
1- appeared argument is when the frame appears x times it saves the image?
2-How can i add yolo arguments such as conf, iou, or save path?

**Reply by dh031200** [reply]
@Ziad-Assem 
1. you're right. If the object appears n or more times, the image is saved. This is to suppress false detection and duplication for the same object.
2. I added code to pass arguments through config.yaml. I'll write a readme soon

**Reply by Ziad-Assem** [reply]
Thank you a lot! I'm new to this and this helped a bunch. I just noticed the config.yaml. I'll definitely read your readme as soon as it's uploaded.

**Reply by glenn-jocher** [reply]
@Ziad-Assem i'm glad to hear that the community's solutions are helping you with your project! Remember, the `appeared` argument is a great way to filter out transient detections by ensuring an object is consistently detected before saving the image. For further customization like confidence thresholds (`conf`), Intersection Over Union (`iou`), or setting a save path, adjusting the parameters in the `config.yaml` file is the right approach. Keep an eye out for the updated readme for detailed instructions. If you have any more questions or need assistance with YOLOv8 specifically, feel free to reach out. Happy detecting! 😊🚀
